swift-sdk/IterableAPI.swift

```diff
@@ -45,6 +45,7 @@ public final class IterableAPI : NSObject {
                                   launchOptions: [UIApplication.LaunchOptionsKey: Any]? = nil,
                                   config: IterableConfig = IterableConfig()) {
         internalImplementation = IterableAPIInternal.initialize(apiKey: apiKey, launchOptions: launchOptions, config:config)
+        NotificationCenter.default.post(name: .iterableAppReady, object: self, userInfo: nil)
```

Comment:
Why the change from `.synchronize` to `NotificationCenter`?

Reply:
Good question. Calling synchronize in initializer was causing some problems because `internalImplementation` was not set properly. This would show up only in unit tests where the app is being initialized more than once. But this is a better implementation than calling synchronize from inside the initializer.

Reply:
`.synchronize` seemed a bit more explicit to me, even though it is a direct call. A `NotificationCenter` message is something that's global and can even be sent by anyone.